dask/array/routines.py

```diff
@@ -531,10 +531,18 @@ def round(a, decimals=0):
 
 @wraps(np.unique)
 def unique(x):
-    name = 'unique-' + x.name
-    dsk = dict(((name, i), (np.unique, key)) for i, key in enumerate(x._keys()))
-    parts = Array._get(sharedict.merge((name, dsk), x.dask), list(dsk.keys()))
-    return np.unique(np.concatenate(parts))
+    x = x.ravel()
+
+    out = atop(np.unique, "i", x, "i", dtype=x.dtype)
+    out._chunks = tuple((np.nan,) * len(c) for c in out.chunks)
+
+    name = 'unique-' + out.name
```

Comment:
This results in something like `unique-unique-<ID>` being added to the graph. I guess that makes some sort of sense, but would be open to better naming suggestions.

Reply:
Perhaps `"unique-aggregate-" + ...`

Reply:
Changed. :)